torch/optim/optimizer.py

```diff
@@ -226,7 +226,7 @@ def __repr__(self):
 
     # Currently needed by Adam and AdamW
     def _cuda_graph_capture_health_check(self):
-        if torch.has_cuda and torch.cuda.is_available():
+        if not is_compiling() and torch.has_cuda and torch.cuda.is_available():
```

Comment:
you don't correctly set `capturing` status when you are compiling? What happens then?

Reply:
So if we're compiling I think I'm going to take the capturable path automatically vs setting the flag

Reply:
Leave a comment about that